### PR TITLE
Add vAPI bindings for vLCM - offline depots and cluster baseline images

### PR DESCRIPTION
**Pull request by stoyanzhelyazkov**
## Description

This PR adds bindings for parts of the vLCM REST API. More specifically for the methods that allow the manipulation of offline depots and of cluster baseline images.

These bindings are necessary for the implementation of vLCM automation in https://github.com/hashicorp/terraform-provider-vsphere

Links to the doc entries for each method included in this change are provided in the source code.

## Type of change

Please mark options that are relevant:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] This change requires a documentation update
- [ ] Build related change

## How Has This Been Tested?

Added handlers in the simulator and a new bats test
I have also tested the new bindings on a live environment. Below you can find a scenario that goes through every new method in order:

Preparation steps
1. Deploy a vCenter server and an ESXi host
2. Move the host into a cluster
3. Upload a software bundle (drivers, vibs, etc.) to a location where it is accessible via http/https

Test steps. Call the following methods
1. "CreateOfflineDepot" - to create a new depot with your software bundle
2. "GetOfflineDepots" - to see the list of depots. Copy the identifier of your new depot
3. [Optional] "GetOfflineDepot" - to see the details of an individual depot
4. "GetOfflineDepotContent" - to obtain the identifier and version of your component (driver/vib/something)
5. [Optional] "ListSoftwareDrafts" - to see if there are any drafts
6. [Optional] "DeleteSoftwareDraft" - to clean up any unfinished drafts
7. "CreateSoftwareDraft" - to open a new baseline draft
8. "UpdateSoftwareDraftComponents" - to add your components to the draft
9. [Optional] "GetSoftwareDraftComponent" - to check if everything is okay in the draft
10. "CommitSoftwareDraft" - to commit and apply your draft
11. "CreateSoftwareDraft" - to create a second draft
12. "RemoveSoftwareDraftComponents" - to remove your component from the image
13. "CommitSoftwareDraft" - to commit and apply your second draft
14. "DeleteOfflineDepot" - to delete the offline depot from step 1

## Checklist:

- [X] My code follows the `CONTRIBUTION` [guidelines] of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

[guidelines]: https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by dougm — COMMENTED** [review]
Thanks @stoyanzhelyazkov 
Looks great overall, just some minor suggestions

File: govc/cluster/draft/component/ls.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	rc, err := cmd.RestClient()
+
+	if err != nil {
+		return err
+	}
+
+	dm := clusters.NewManager(rc)
+
+	var d map[string]clusters.SettingsComponentInfo
+	if d, err = dm.ListSoftwareDraftComponents(cmd.clusterId, cmd.draftId); err != nil {
```
[the remark this answers is not in the document]
Comment (stoyanzhelyazkov) [reply govc/cluster/draft/component/ls.go:102]: If I use := to declare & assign the result variable "d" would be bound to the conditional block.
Which would mean that will have to add an else block for the "success case"

I just prefer having an unconditional return at the end of my functions.

Of course I will comply with the style of the repository :) just wanted to provide some justification

**Review by dougm — COMMENTED** [review]
Awesome, thanks @stoyanzhelyazkov 
If you can just [squash](https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md#updating-pull-requests) the commits, it can be merged

(multiple commits are fine, just want to squash the fixup commits)

**Comment by stoyanzhelyazkov** [comment]
> Awesome, thanks @stoyanzhelyazkov If you can just [squash](https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md#updating-pull-requests) the commits, it can be merged
> 
> (multiple commits are fine, just want to squash the fixup commits)

Squashed everything into one commit

**Comment by stoyanzhelyazkov** [comment]
@dougm Please don't merge yet.
I found a better API endpoint for one of the workflows. I want to make some more changes.

**Review by dougm — APPROVED** [review]
Thanks again @stoyanzhelyazkov

**Pull request merged**